source/docs/prismic/routable-types.md

```diff
@@ -0,0 +1,231 @@
+
+2- Select the custom type to make routable
+
+3- Drag and drop "Key Text - Text field for exact match search" from the right to the Main tab on the left.
+
+4- Give the new field a "Field name", "API ID" and optionally a "Field placeholder" (note the "API ID" is an important field and will be used below)
+
+5- Click ok to confirm the inputs
+
+6- Save the changes
+
+**For the sake of simplicity we will assume the name of this field is `url` for the rest of this documentation**
+
+## Create a GraphQL type to represent the custom data type
+
+If no GraphQL type was created a to represent the custom data type its time to do so.
+
+Create a GraphQL type to represent the custom datatype in schema.gql of the server module's directory
+
+```diff
++type Album implements Routable {
++  url: String # the newly added field
++  artist: String
++  title: String
++  release_date: Date
++  path: String # for the routable interface
++}
+```
+
+Note that inorder for the custom type to be routable it needs to implement the core front-commerce [Routable interface](https://gitlab.com/front-commerce/front-commerce/-/blob/2.7.1/src/server/modules/front-commerce/core/schema.gql#L87). Which basically means the GraphQL type will have to have a field called `path` to represent the url of the routable type.
+
+Add a new GraphQL type to the module's resolver:
+
+```diff
+export default {
+  Query: {
+    ...
+  },
++  Album: {
++    path: (content) => {
++      return (
++        "/" + content.url
++      );
++    },
++  },
++};
+```
+
+Note we added a path field resolver that appends a `/` to the `url` field of our custom prismic type.
+
+## Register the prismic custom type as a routable type
+
+To register the custom type as a routeable prismic type it must be registered with the prismic module. First ensure that the `Prismic/core` is added as a module's dependency then use `loaders.Prismic.registerRoutableType` to add the custom type as routable type:
+
+```diff
+import typeDefs from "./schema.gql";
+import resolvers from "./resolvers";
+
+export default {
+  namespace: "<module_name_space>",
+-  dependencies: [],
++  dependencies: ["Prismic/Core"],
+  typeDefs,
+  resolvers,
+  contextEnhancer: ({ req, loaders }) => {
+
++    const { TitleTransformer, DateTransformer } = loaders.Prismic.transformers;
++    const contentTransformOptions = {
++      fieldTransformers: {
++        title: new TitleTransformer(),
++        release_date: new DateTransformer(),
++      },
++    };
++
++    loaders.Prismic.registerRoutableType({
++      typeIdentifier: "album", // <-- prismic custom type
++      urlFieldName: "url",     // <-- prisic API ID mentioned above
++      graphQLType: "Album",    // <-- GraphQL type created above
++      contentTransformOptions,
++      isSitemapable: false,
++      postTransformer: (url, document) => { // optional function postTransformer
```

Comment:
it needs to be detailed a bit somewhere (what is `url` and `document` + return value and behaviour)